plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java

```diff
@@ -817,4 +922,202 @@ private Map<String, String> getTableProperties(String schemaName, String tableNa
             return properties.buildOrThrow();
         }
     }
+
+    @Test
+    @Override
+    public void testSetColumnType()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA));
+        String tableName = "test_set_column_type";
+        assertUpdate("CREATE TABLE " + tableName + " (a bigint, b double NOT NULL, c varchar(50)) WITH (order_by=ARRAY['b'], engine = 'MergeTree')");
+        assertUpdate("ALTER TABLE " + tableName + " ALTER COLUMN a SET DATA TYPE varchar(50)");
+        assertEquals(getColumnType(tableName, "a"), "varchar");
+        assertThat((String) computeScalar("show create table " + tableName)).contains("CREATE TABLE clickhouse.tpch.test_set_column_type (\n" +
+                "   a varchar NOT NULL,\n" +
+                "   b double NOT NULL,\n" +
+                "   c varchar\n" +
+                ")\n" +
+                "WITH (\n" +
+                "   engine = 'MERGETREE',\n" +
+                "   order_by = ARRAY['b'],\n" +
+                "   primary_key = ARRAY['b']\n" +
+                ")");
+    }
+
+    @Test
+    @Override
+    public void testSetColumnIncompatibleType()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_SET_COLUMN_TYPE) && hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA));
+
+        String tableName = "test_set_column_incompatible_type";
+
+        assertUpdate("CREATE TABLE " + tableName + " (col bigint, col2 int not null) WITH (order_by=ARRAY['col2'], engine = 'MergeTree')");
+        assertUpdate("ALTER TABLE " + tableName + " ALTER COLUMN col SET DATA TYPE integer");
+        assertEquals(getColumnType(tableName, "col"), "integer");
+    }
+
+    @Test
+    @Override
+    public void testSetColumnOutOfRangeType()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_SET_COLUMN_TYPE) && hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA));
+
+        String tableName = "test_set_column_out_of_range";
+        assertUpdate("CREATE TABLE " + tableName + " (col bigint, col2 int not null) WITH (order_by=ARRAY['col2'], engine = 'MergeTree')");
+        query("insert into " + tableName + " values(9223372036854775807, 22)");
+        assertUpdate("ALTER TABLE " + tableName + " ALTER COLUMN col SET DATA TYPE integer");
+
+        assertEquals(getColumnType(tableName, "col"), "integer");
+        assertQuery("SELECT col FROM " + tableName, "VALUES -1");
```

Comment:
This looks like a bug. We may need to deny such operations. Could you find a GitHub issue of ClickHouse about this?

Reply:
`int` seems to map to `Int32` in ClickHouse, the max value in ClickHouse for `Int32` is `2147483647`.
```
┌─statement──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ CREATE TABLE tpch.test_set_column_out_of_rangekgarmodrdz
(
    `col` Int32,
    `col2` Int32
)
ENGINE = MergeTree
ORDER BY col2
SETTINGS index_granularity = 8192 │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘


```

Reply:
Let's deny changing such types.